build/crds/router/router_v1_rule.yaml

```diff
@@ -0,0 +1,65 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: rules.rules.kubeedge.io
+spec:
+  group: rules.kubeedge.io
+  versions:
+    - name: v1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                source:
+                  description: |
+                    source is a string value representing where the messages come from. Its
+                    value is the same with rule-endpoint name. For example, my-rest or my-eventbus.
+                  type: string
+                sourceResource:
+                  description: |
+                    sourceResource is a map representing the resource info of source. For rest
+                    rule-endpoint type its value is {"path":"/a/b"}. For eventbus rule-endpoint type its
+                    value is {"topic":"<user define string>","node_name":"xxxx"}
+                  type: object
+                  additionalProperties:
+                    type: string
+                target:
+                  description: |
+                    target is a string value representing where the messages go to. its value is
+                    the same with rule-endpoint name. For example, my-eventbus or my-rest or my-servicebus.
+                  type: string
+                targetResource:
+                  description: |
+                    targetResource is a map representing the resource info of target. For rest
+                    rule-endpoint type its value is {"resource":"http://a.com"}. For eventbus rule-endpoint
+                    type its value is {"topic":"/xxxx"}. For servicebus rule-endpoint type its value is
+                    {"path":"/request_path"}.
+                  type: object
```

Comment:
better to define a object for eventbus/rest with explict property:
eventbus =>
```yaml
eventbusResource:
  type: object
  properties:
    topic:
      type: string
```

rest =>
```yaml
restResource:
  type: object
  properties:
    path:
      type: string
```

then use openapi (oneOf)[https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/)

Reply:
users can refer to the description of properties and we use webhook to check the properties.